modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1249 @@
+      ``a.insert((a.size), x)`` is equivalent to ``a.append(x)``.
+
+      If the insertion is successful, this method returns `true`. If the given
+      index is out of bounds, this method does nothing and returns `false`.
+
+      .. warning::
+      
+        Inserting an element into this unrolledLinkedList may invalidate existing references
+        to the elements contained in this unrolledLinkedList.
+
+      :arg idx: The index into this unrolledLinkedList at which to insert.
+      :type idx: `int`
+
+      :arg x: The element to insert.
+      :type x: `eltType`
+
+      :return: `true` if `x` was inserted, `false` otherwise.
+      :rtype: `bool`
+    */
+    proc ref insert(idx: int, x: eltType): bool
+         lifetime this < x {
+      var result = false;
+
+      on this {
+        _enter();
+
+        // Handle special case of `a.insert((a.size), x)` here.
+        if idx == _size {
+          _append(x);
+          result = true;
+        } else if _withinBounds(idx) {
+          _insert(idx, x);
+          result = true;
+        }
+
+        _leave();
+      }
+
+      return result;  
+    }
+
+    pragma "no doc"
+    proc ref _insertGenericKnownSize(idx: int, items, size: int): bool {
+      //TODO: Maybe some optimization here. This O(N^2)
+      var result = false;
+
+      _sanity(size >= 0);
+
+      if size == 0 then
+        return true;
+
+
+      on this {
+        if idx == _size {
+          for x in items {
+            _append(x);
+          }
+          result = true;
+        }
+        else if _withinBounds(idx) {
+          var count = 0;
+          for x in items {
+            _insert(idx + count, x);
+            count += 1;
+          }
+          result = true;
+        }
+      }
+
+      return result;
+    }
+
+    /*
+      Insert an array of elements `arr` into this unrolledLinkedList at index `idx`,
+      shifting all elements at and following the index `arr.size` positions
+      to the right. 
+
+      If the insertion is successful, this method returns `true`. If the given
+      index is out of bounds, this method does nothing and returns `false`.
+
+      .. warning::
+
+        Inserting elements into this unrolledLinkedList may invalidate existing references
+        to the elements contained in this unrolledLinkedList.
+
+      :arg idx: The index into this unrolledLinkedList at which to insert.
+      :type idx: `int`
+
+      :arg arr: An array of elements to insert.
+      :type x: `[] eltType`
+
+      :return: `true` if `arr` was inserted, `false` otherwise.
+      :rtype: `bool`
+    */
+    proc ref insert(idx: int, arr: [?d] eltType): bool lifetime this < arr {
+
+      var result = false;
+
+      on this {
+        _enter();
+        result = _insertGenericKnownSize(idx, arr, arr.size);
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Insert a list of elements `lst` into this unrolledLinkedList at index `idx`, shifting
```

Comment:
Same here. Make it clear that we are inserting all the elements in the list

Reply:
(and I am not checking the List docs, you may have inherited this language from there. Still, would like this to change, and would like the List doc to change, too, if that's the case :) )